src/plugins/ucx/ucx_backend.cpp

```diff
@@ -1623,13 +1629,80 @@ nixl_status_t nixlUcxEngine::releaseReqH(nixlBackendReqH* handle) const
 }
 
 nixl_status_t
-nixlUcxEngine::createGpuXferReq(const nixlBackendReqH &handle,
+nixlUcxEngine::createGpuXferReq(const nixlBackendReqH &req_hndl,
+                                const nixl_meta_dlist_t &local_descs,
+                                const nixl_meta_dlist_t &remote_descs,
                                 nixlGpuXferReqH &gpu_req_hndl) const {
+#ifdef HAVE_UCX_GPU_DEVICE_API
```

Comment:
We can get rid of this `ifdef` in this file. We can place the `ifdef` into `deviceMemList::createDeviceMemList` to use the UCX device API on `utils/ucx` level (as we did for other UCX device APIs).